### PR TITLE
make scalar clamp overloads propagate nan

### PR DESCRIPTION
**Pull request by ngimel**

### CODE REVIEW

File: aten/src/ATen/native/TensorCompare.cpp [hunk 1]
```diff
@@ -555,10 +560,15 @@ TORCH_IMPL_FUNC(clamp_out)
  const Tensor& /*self*/,
  const OptionalScalarRef min,
  const OptionalScalarRef max,
- const Tensor& /*result*/) {
+ const Tensor& result) {
   using at::native::detail::ClampLimits;
   if (min && max) {
-    clamp_scalar_stub(device_type(), *this, min.get(), max.get());
+    if (min.get().toDouble() != min.get().toDouble() ||
```
Comment (mruberry) [inline aten/src/ATen/native/TensorCompare.cpp:566]: Cool short-circuit

File: aten/src/ATen/native/TensorCompare.cpp [hunk 2]
```diff
@@ -580,7 +590,14 @@ TORCH_IMPL_FUNC(clamp_Tensor_out)
 
 TORCH_IMPL_FUNC(clamp_max_out)
 (const Tensor& self, const Scalar& max, const Tensor& result) {
-  clamp_max_scalar_stub(device_type(), *this, max);
+  if (max.toDouble() != max.toDouble()) {
+//TODO this is not great, building TI again is expensive, but I can't use
```
Comment (mruberry) [inline aten/src/ATen/native/TensorCompare.cpp:594]: Nice comment

**Review by mruberry — APPROVED** [review]
Cool -- let me know if you want to add a NaN propagation test -- I have a draft for one

**Comment by ngimel** [comment]
I think we can add nan propagation tests together with clamp decompositions.

**Pull request closed without merging**